agent/containers/images/image-test-script.sh

```diff
@@ -75,14 +75,22 @@ export pbench_run=$PBENCH_RUN_DEFAULT
 while [[ -d $pbench_run ]]; do
     read -p "Is it ok to delete the contents of '$pbench_run'? (y/[n]): " delete_check
     delete_check=$(echo $delete_check | tr '[:upper:]' '[:lower:]')
-    if [[ -z $delete_check || ($delete_check != 'y' && $delete_check != "yes") ]]; then
-        read -p "Suggest a new alternate directory for the pbench run (will be wiped/created): " pbench_run
-        export pbench_run
+    if [[ "$delete_check" != 'y' && "$delete_check" != "yes" ]]; then
+        while :; do
+            read -p "Suggest a new alternate directory for the pbench run (will be wiped/created): " pbench_run
+            parent="$(dirname "$pbench_run")"
+            if [ -d $parent ]; then
+                export pbench_run
+                break
+            else
+                printf -- "Invalid dir '$pbench_run', cannot be created\n"
+            fi
+        done
     else
+        rm -rf ${pbench_run}
         break
     fi
 done
-rm -rf ${pbench_run}
 mkdir ${pbench_run}
```

Comment:
You shouldn't need the `break` statement at line 91 -- the `while` test at line 75 will let control out of the loop if the `rm` succeeded (and, if it didn't succeed, then you probably would like to loop again...).  And, therefore, I don't think you need the `else` at line 89, either.  And, I'm not sure that  the interior `while` loop is needed or does what you want.

Would something like the following suffice?
```
while [[ -d $pbench_run ]]; do
    read -p "Is it ok to delete the contents of '$pbench_run'? (y/[n]): " delete_check
    delete_check=$(echo $delete_check | tr '[:upper:]' '[:lower:]')
    if [[ "$delete_check" == 'y' || "$delete_check" == "yes" ]]; then
        rm -rf ${pbench_run}
    else
        read -p "Suggest a new alternate directory for the pbench run (will be wiped/created): " pbench_run
done
export pbench_run
mkdir ${pbench_run}
```
This doesn't handle failures to create `${pbench_run}`, but...how paranoid do we want to be?

Reply:
The internal while loop is there to ensure that any specified dir for creation has an existing parent. You are right about the else and break though, and this can be cut down a bit.

Reply:
Actually I've re-added the else, since I want to prompt the user again for an extra layer of protection (before deleting an existing dir)

Reply:
> to ensure that any specified dir for creation has an existing parent

This isn't strictly necessary; and, if you add the `-p` switch to the `mkdir` command, then it will automatically create the parent(s), if necessary.

> I've re-added the else, since I want to prompt the user again for an extra layer of protection

The code that I proposed also does that:  if the user responds to the "alternate directory" prompt, then control returns to the top of the loop; if the alternate directory exists, the script will prompt before attempting to delete it.

Reply:
This is a great point. Using your suggestion, I've significantly simplified the loop, thanks!